### DISCUSSION TITLE
Use a ref struct for local function captures.

### DISCUSSION THREAD

**Discussion by ThadHouse** [dotnet/csharplang#1809] [Language Ideas]
Since the local function captures can't escape, it seems like they should just be able to be ref structs. By doing this, capturing spans into local functions would be allowed, which would be nice. I can't see any reason this would break any existing code, since you can't get direct access to the capture variable anyway.

I tried looking for an existing issue about this, and couldn't find one.

**Comment by agocke** [comment]
Local functions can be converted to delegates, so they can escape. If the compiler can prove they never get captured or converted to a delegate, then it's known that they can't escape.

I'm not sure this question is specific to Span, though. The same rules apply for all `ref` variables.
